src/openrct2/scripting/ScPark.hpp

```diff
@@ -280,11 +280,20 @@ namespace OpenRCT2::Scripting
             context_broadcast_intent(&intent);
         }
 
+        money16 entranceFee_get() const
+        {
+            return gParkEntranceFee;
+        }
+        void entranceFee_set(money16 value)
+        {
+            ThrowIfGameStateNotMutable();
+            park_set_entrance_fee(value);
```

Comment:
This is calling a game action. @intelOrca should this be instead be performed through the plugin game actions code and using the visit function.

Reply:
I don't have anything against having an `entranceFee_set`, it will only be able to be used when game state is mutable such as in a custom action or certain hooks, but it should not be performing a game action. It should just set the price directly.

Reply:
If it sets it directly then you'll have to duplicate all the validation code and it won't work in networked games. Or is that what the mutable protection ensures.

Reply:
The mutable thing already protects it from network issues, you use the game action if you want network sync. The validation can still be used, just extract that out to another function you can call.

Reply:
I'll look into that. Out of curiosity: Is calling a game action from the plugin API generally not a good practice? Why so?

Reply:
Game actions are asynchronous, where as a plugin would expect this property to be changed and updated immediately.

Reply:
Ah, I see. Other setters here either just copy the boundary checks from other places (e.g. `rating_set`) or don't perform any kind of validation/sanitization. Any suggestions for where to put a shared validation method and how to structure it? It has to be more than a single bool check because currently [`SetParkEntranceFeeAction`](https://github.com/OpenRCT2/OpenRCT2/blob/22f29bf192320ef78cabd8bc76672f1e85cad038/src/openrct2/actions/SetParkEntranceFeeAction.hpp#L47-L56) differentiates the "no entrance fee allowed" and "entrance fee too high/low" cases.

Reply:
Probably just `Park.cpp`

Reply:
We don't really want to get into the same issue as the ride price setting value issues so I think we still need to check for the no entrance fee allowed.

Reply:
But if you toggle that park flag you would probably expect it to keep the price it had before you toggled it. We just need to make sure we use the get method for everything that uses the price so we can return free if it is disabled.

Reply:
Hadn't even noticed that. I should probably use `park_get_entrance_fee();` in the getter?

Reply:
Like I keep saying, this is supposed to represent the raw property, so I really don't think there is any need for disguise or validation.

Reply:
Yeah, I guess it's a question of philosophy. If it should just allow access to the raw values (like the other fields), I'll just drop the validation for now and update `gParkEntranceFee` directly.

Reply:
Updated the PR to just allow direct access to `gParkEntranceFee` like @IntelOrca suggested.